### PR TITLE
Fix issue #884

### PR DESCRIPTION
**Pull request by rparthas**
Provide method to reset intercept for a given url
Provide method to reset intercept for all url
https://github.com/getgauge/taiko/issues/884

### CODE REVIEW

File: lib/taiko.js [hunk 1]
```diff
@@ -3704,3 +3706,27 @@ module.exports.metadata = {
...
+    networkHandler.resetInterceptor(requestUrl);
+    descEvent.emit(
+      'success',
+      'Intercepts reset for url' + requestUrl,
+    );
+  } else {
+    networkHandler.resetInterceptors();
+    descEvent.emit('success', 'Intercepts reset for all url');
+  }
+};
```
[the remark this answers is not in the document]
Comment (rparthas) [reply lib/taiko.js]: Sure Ok. Will address of the comments

**Comment by rparthas** [comment]
@negiDharmendra can you please review the pull request

File: lib/taiko.js [hunk 2]
```diff
@@ -3725,3 +3726,29 @@ module.exports.metadata = {
     'repl',
   ],
 };
+
+/**
+ * Add interceptor for the network call. Helps in overriding request or to mock response of a network call.
```
Comment (negiDharmendra) [inline lib/taiko.js]: Please update the docs content.

File: lib/taiko.js [hunk 3]
```diff
@@ -3725,3 +3726,29 @@ module.exports.metadata = {
...
+module.exports.resetIntercept = requestUrl => {
+  if (requestUrl) {
+    var success = networkHandler.resetInterceptor(requestUrl);
+    if (success) {
+      descEvent.emit(
+        'success',
+        'Intercepts reset for url ' + requestUrl,
+      );
+    }
+  } else {
```
Comment (negiDharmendra) [inline lib/taiko.js]: I think `resetIntercept` should inform the user when there is no intercept available for the provided `requestUrl`

**Comment by negiDharmendra** [comment]
Resolve the conflicts in `api.json`

File: lib/handlers/networkHandler.js [hunk 4]
```diff
@@ -61,19 +61,23 @@ const responseHandler = response => {
   eventHandler.emit('responseReceived', response);
 };
 
+const getMatchingInterceptor = (interceptor, url) => {
```
Comment (zabil) [inline lib/handlers/networkHandler.js]: Sorry for the late review comment.
Can we rename this method to `clearIntercept`?

Comment (rparthas) [reply lib/handlers/networkHandler.js]: @zabil I don't think the method name would be correct. It just identifies the matching interceptor

Comment (zabil) [reply lib/handlers/networkHandler.js]: I meant renaming `resetInterceptor` to `clearInterceptor`

Comment (rparthas) [reply lib/handlers/networkHandler.js]: I renamed resetIntercept to clearIntercept if thats what you were referring to .

**Comment by negiDharmendra** [comment]
@rparthas Thanks for your valuable contribution to Taiko.

**Pull request merged**